### DISCUSSION TITLE
Testing of functions in libsodium

### DISCUSSION THREAD

**Discussion by arindam-8** [jedisct1/libsodium#1337] [Q&A]
Hi,

I came across this issue with libsodium while working with it and saw that a couple of functions are not being tested by the project’s regression test suite. For instance, `blake2b_compress_ref()` function is not touched by any of the tests in the regression test suite since it doesn’t get picked up by the function that selects the “best implementation” based on macros like `HAVE_EMMINTRIN_H`. This happens in a couple other cases as well.
Hence I was wondering if there is some other mechanism by which those functions are being tested.

Thanks
Arindam

**Comment by jedisct1** [comment]
By testing on different architectures.
